src/main/java/delete_post/drivers/DeletePostDataAccess.java

```diff
@@ -1,25 +1,118 @@
+        this.postPath = postPath;
+        this.userPath = userPath;
+        this.postReader = postReader;
     }
 
     @Override
-    public void removeFavourite(int postId, int userId){
+    public Post getPost(int postId){
+        List<String[]> postData = readFile(postPath);
+        for (String[] row : postData) {
+            if (Integer.parseInt(row[0]) == postId) {
+                return postReader.readPost(row);
+            }
+        }
+        System.out.println("Could not find post in database");
+        return null;
+    }
 
+    public void removeFavourite(int postId, int userId){
+        removeFromUserList(postId, userId, 1);
     }
-    public void removeTag(int postId, String tag){
 
+    public void removeUser(int postId, int userId){
+        removeFromUserList(postId, userId, 2);
     }
+
     public void deletePost(int postId){
 
+        List<String[]> postData = readFile(postPath);
+
+        for (String[] row: postData){
+            if (Integer.parseInt(row[0]) == postId){
+                for (int i=1; i<9; i++){
+                    row[i] = null;
+                }
+                break;
+            }
+        }
+
+        fileWriter(postPath, postData);
+    }
+
+    private List<String[]> readFile(String path) {
+        File file = new File(path);
+        List<String[]> data = null;
+        try {
+            // read
+            FileReader fileReader = new FileReader(file);
+            CSVReaderBuilder csvReaderBuilder = new CSVReaderBuilder(fileReader);
+
+            CSVReader reader = csvReaderBuilder.build();
+            data = reader.readAll();
+            // Remove the row corresponding to the header
+            data.remove(0);
+        } catch (IOException e) {
+            System.out.println("Error during accessing file.");
+        } catch (CsvException e) {
+            System.out.println("Error during reading file");
+        }
+        return data;
+    }
+
+    private void fileWriter(String path, List<String[]> data) {
+        try {
+            FileWriter newFile = new FileWriter(path);
+            CSVWriter writer = new CSVWriter(newFile);
+            writer.writeAll(data);
+            writer.close();
+        } catch (IOException e) {
+            System.out.println("Error during writing file.");
+        }
+    }
+
+    private void removeFromUserList(int postId, int userId, int listCol){
+        List<String[]> userData = readFile(userPath);
+
+        for (String[] row: userData){
+            if (userId == Integer.parseInt(row[0])){
+                StringBuilder newList = new StringBuilder();
+                String[] oldList = row[listCol].split(" ");
+                for (String id : oldList){
+                    if (Integer.parseInt(id) != postId) {
+                        if (newList.length() != 0){
+                            newList.append(" ");
+                        }
+                        newList.append(id);
+                    }
+                }
```

Comment:
I think when you run this, it will add a space at the end of the newList. So then you'll have a StringBuilder like "1 2 3 ", which may be a problem for other use cases. Can you remove the last space in the StringBuilder? Except if newList is empty or just of length 1 so if it is like "" or "1".